src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Tree/Visitors/ExpressionTreeRewriter.cs

```diff
@@ -104,7 +104,7 @@ protected override Expr VisitBOUNDLAMBDA(ExprBoundLambda anonmeth)
 
             CType delegateType = anonmeth.DelegateType;
             TypeArray lambdaTypeParams = GetSymbolLoader().getBSymmgr().AllocParams(1, new CType[] { delegateType });
-            AggregateType expressionType = GetSymbolLoader().GetOptPredefTypeErr(PredefinedType.PT_EXPRESSION);
+            AggregateType expressionType = GetSymbolLoader().GetPredefindType(PredefinedType.PT_EXPRESSION);
```

Comment:
Static compiler needs to deal with cases where it is compiling against a runtime that is not the runtime it is running on. That is not a case for the dynamic binder.

---
In reply to: [129954237](https://github.com/dotnet/corefx/pull/22471#discussion_r129954237) [](ancestors = 129954237)